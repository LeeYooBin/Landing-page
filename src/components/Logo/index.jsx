import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Content = styled.div`
  margin: 0;
  padding: 0;
  width: auto;
  height: auto;
`;

const Link = styled.a`
  width: auto;
  height: auto;
`;

const Image = styled.img`
  width: 20.1rem;
  height: 4.124rem;
`;

const Logo = ({ image, isClickable = false, href = '#' }) => {
  if (!image) {
    throw new Error('The prop `image` is required in Logo component.');
  }

  return (
    <Content>
      {isClickable ? (
        <Link href={href}>
          <Image src={image} alt="Logo" />
        </Link>
      ) : (
        <Image src={image} alt="Logo" />
      )}
    </Content>
  );
};

Logo.propTypes = {
  image: PropTypes.string.isRequired,
  isClickable: PropTypes.bool,
  href: PropTypes.string,
};

export default Logo;
