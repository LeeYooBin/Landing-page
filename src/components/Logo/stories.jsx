import Logo from '.';

export default {
  title: 'Logo',
  component: Logo,
};

export const Template = (args) => <Logo {...args} />;
