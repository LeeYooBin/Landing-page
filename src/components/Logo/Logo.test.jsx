import { describe, it, expect } from '@jest/globals';
import { screen } from '@testing-library/react';
import Logo from '.';
import renderTheme from '../../styles/render-theme';

describe('Logo component', () => {
  it('Should render without errors', () => {
    renderTheme(
      <Logo image="logo.png" href="https://example.com" isClickable />,
    );
  });

  it('renders image correctly', () => {
    renderTheme(<Logo image="logo.png" />);
    const image = screen.getByRole('img');
    expect(image).toBeInTheDocument();
    expect(image.src).toContain('logo.png');
  });

  it('renders clickable link when isClickable is true', () => {
    renderTheme(
      <Logo image="logo.png" href="https://example.com" isClickable />,
    );
    const linkElement = screen.getByRole('link');
    expect(linkElement).toBeInTheDocument();
    expect(linkElement.href).toContain('https://example.com');
  });

  it('renders non-clickable image when isClickable is false', () => {
    renderTheme(<Logo image="logo.png" />);
    expect(screen.getByRole('img')).toBeInTheDocument();
    expect(screen.queryByRole('link')).not.toBeInTheDocument();
  });

  it('throws an error when image prop is missing', () => {
    expect(() => {
      renderTheme(<Logo />);
    }).toThrowError();
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(
      <Logo image="logo.png" href="https://example.com" isClickable />,
    );
    expect(container).toMatchSnapshot();
  });
});
