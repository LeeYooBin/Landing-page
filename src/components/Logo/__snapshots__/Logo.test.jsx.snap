// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component should match snapshot 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  width: auto;
  height: auto;
}

.c1 {
  width: auto;
  height: auto;
}

.c2 {
  width: 20.1rem;
  height: 4.124rem;
}

<div>
  <div
    class="c0"
  >
    <a
      class="c1"
      href="https://example.com"
    >
      <img
        alt="Logo"
        class="c2"
        src="logo.png"
      />
    </a>
  </div>
</div>
`;
